Barotrauma/BarotraumaShared/SharedSource/Events/Missions/GoToMission.cs

```diff
@@ -4,9 +4,13 @@ namespace Barotrauma
 {
     partial class GoToMission : Mission
     {
+        private readonly bool maxProgressStateDeterminsCompleted;
+        private readonly int failState;
         public GoToMission(MissionPrefab prefab, Location[] locations, Submarine sub)
             : base(prefab, locations, sub)
         {
+            maxProgressStateDeterminsCompleted = prefab.ConfigElement.GetAttributeBool("maxprogressdeterminescompleted", false);
+            failState = prefab.ConfigElement.GetAttributeInt("failstate", -1);
```

Comment:
The mission implicitly fails if it's in the fail state as it's not in the success state at the end of the round, unless the state is changed afterwards in which case it will no longer be in the fail state and will thus not display as failed. Unless both the success state and fail state are the same, which imo would be a fault of the mission creator but I could put in a warning for it just to catch mistakes.

Reply:
In the current impl presenting by this pull request you could have a fail state that's higher than the MaxProgressState which would lead to weird situations where the mission completes even though it's displayed as failed, however the upcoming changes will fix that.